command-line/lib/Command/Address/Delegation.hs

```diff
@@ -77,7 +79,7 @@ run Cmd{credential} = do
         Left (ErrInvalidAddressType  msg) ->
             fail msg
         Right addr ->
-            B8.hPutStr stdout $ T.encodeUtf8 $ bech32With hrp addr
+           B8.hPutStr stdout $ T.encodeUtf8 $ bech32With hrp (unAddress addr)
```

Comment:
Shouldn't this be using ToBech32 instead of unwrapping the Address?

Reply:
It can contain two different prefixes, so no. I haven't yet though about how to handle prefix hierarchies (here both `addr` and `addr_test`).

Reply:
I do agree with Rodney here, if we can't use the abstraction in places where it would make sense then it is of little use :no_mouth: 

However, there's enough information already in an address to know which prefix should be used. The first byte of an address does contain network information. 

See: https://github.com/input-output-hk/cardano-ledger-specs/blob/master/shelley-ma/shelley-ma-test/cddl-files/shelley-ma.cddl#L71-L79

Reply:
> However, there's enough information already in an address to know which prefix should be used. The first byte of an address does contain network information.

That _would_ have been nice, but the stake keys don't contain the network id, only normal addresses

Reply:
Stake addresses do contain a network id. And prefixes is only different between testnet and mainnet for stake addresses, not stake keys.

Reply:
`stake` /= `stake_vk`

Reply:
> Stake addresses do contain a network id

Ah, ok!